### PR TITLE
feat: Cherry-pick: faster wiring backpressure

### LINKED ISSUE #11689: Faster Back Pressure Implementation
We've observed situations where the wiring framework backpressure is slow under high contention. Make it faster.

### PR DESCRIPTION
**Pull request by netopyr**
**Description**:

Cherry-pick of #11690 

**Related issue(s)**:

Fixes #11689

### CODE REVIEW

**Review by cody-littley — APPROVED** [review]
LGTM. (It helps that I'm the original author. 😜 )

**Comment by cody-littley** [comment]
There was a unit test failure. This change made the tests a bit flaky. I've since fixed a handful of different flakes in `develop`. How should we handle this in the release branch?

File: platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/counters/BackpressureBlocker.java [hunk 1]
```diff
@@ -90,23 +90,15 @@ public boolean block() throws InterruptedException {
...
-            final boolean success = count.compareAndSet(currentCount, currentCount + 1);
-            if (success) {
-                // We've successfully incremented the count, so we're done.
-                return true;
-            }
-
-            // We were unable to increment the count because another thread concurrently modified it.
-            // Try again. We will keep trying until we are either successful or we observe there is
-            // insufficient capacity.
+        final long resultingCount = count.incrementAndGet();
```
Comment (kfa-aguda) [inline platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/counters/BackpressureBlocker.java:93]: I think these changes will benefit from introducing simple unit tests. At the very least, something that asserts the old `isReleasable` and new are the same should suffice.

Comment (netopyr) [reply platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/counters/BackpressureBlocker.java:93]: Should be considered for `develop`.

**Pull request merged**